chain/ethereum/src/network.rs

```diff
@@ -28,15 +33,43 @@ impl EthereumNetworkAdapter {
     fn is_call_only(&self) -> bool {
         self.adapter.is_call_only()
     }
+
+    pub fn get_capacity(&self) -> AvailableCapacity {
+        self.limit.get_capacity(Arc::strong_count(&self.adapter))
+    }
+
+    pub fn current_error_count(&self) -> u64 {
+        self.endpoint_metrics
+            .get_count(&self.adapter.url.as_str().into())
```

Comment:
Can we use `provider` instead to identify the adapter, just to keep the url private?

Reply:
for adapters we can, it leaks a bit of the impl details onto config, added comments where relevant